src/sqlfluff/rules/L025.py

```diff
@@ -62,22 +63,18 @@ def _lint_references_and_aliases(
         for alias in table_aliases:
             if alias.aliased and alias.ref_str not in tbl_refs:
                 fixes = [LintFix.delete(alias.alias_expression)]
-                found_alias_segment = False
                 # Walk back to remove indents/whitespaces
-                for segment in reversed(alias.from_expression_element.segments):
-                    if not found_alias_segment:
-                        if segment is alias.alias_expression:
-                            found_alias_segment = True
-                    else:
-                        if (
-                            segment.name == "whitespace"
-                            or segment.name == "newline"
-                            or segment.is_meta
-                        ):
-                            fixes.append(LintFix.delete(segment))
-                        else:
-                            # Stop once we reach an other, "regular" segment.
-                            break
+                to_delete = (
+                    Segments(*alias.from_expression_element.segments)
+                    .reversed()
+                    .select(
+                        start_seg=alias.alias_expression,
+                        # Stop once we reach an other, "regular" segment.
+                        loop_while=lambda seg: seg.name in ("whitespace", "newline")
+                        or seg.is_meta,
```

Comment:
```suggestion
                        loop_while=sp.or_(sp.is_whitespace, sp.is_meta),
```

Reply:
I'll make this change. There are some subtle differences with some of the whitespace checks (spaces & tabs, or also newlines), but I confirmed this works. 👍

Reply:
Yeah is_whitespace includes newlines 😊